### PR TITLE
logging script

### PR DESCRIPTION
**Pull request by AbiHaddon**
Script for logging 
meets the following requirements:
- appropriate logging levels must be used i.e. debug, info, warning, error
 - Need to log unexpected exceptions
 - Logging is directed to the stream as well as a log file
 - Log level for each is selectable
 - log file is replaced by a rotating log file to prevent the disk from filling

### CODE REVIEW

File: src/logging_1.0.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import logging
```
Comment (AmyGrimwood) [inline src/logging_1.0.py:1]: Needs a docstring at start of script

File: src/logging_1.0.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import sys
+from logging.handlers import RotatingFileHandler
+
+
+class LoggerManager:
```
Comment (AmyGrimwood) [inline src/logging_1.0.py:6]: Needs a docstring documenting function of this class

**Review by AmyGrimwood — CHANGES_REQUESTED** [review]

File: src/logging_1.0.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    def __init__(self, log_file_path='your_log_file.log'):
+        # Create a logger with the name of the current module
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logging.DEBUG)
+
+        # Define a formatter with the desired log message format
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(module)s - %(levelname)s - %(message)s - %(lineno)d',
```
Comment (Fr3dMer) [inline src/logging_1.0.py:13]: Over 79 char long

**Review by Fr3dMer — CHANGES_REQUESTED** [review]
Looks good, just needs unit testing now

File: src/logging_1.0.py (hunk 3 above)
Comment (AbiHaddon) [reply src/logging_1.0.py]: Amended line length

**Comment by AbiHaddon** [comment]
Added doc strings, checked script for pep8 compliance and added exception handling

**Comment by AbiHaddon** [comment]
Updated script with change requests.
Added doc string, exception handling and checked for pep8 compliance

File: src/logging_1.0.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    The LoggerManager class for the simple configuration
+    of a logging system with a rotating file handler and a
+    stream handler.
+
+    Attributes:
+        logger (logging.logger): The logger object for recording
+        log messages.
+    """
+
+    def __init__(self, log_file_path='your_log_file.log'):
```
Comment (Fr3dMer) [inline src/logging_1.0.py:23]: Make the default file something more appropriate for script?

File: src/logging_1.0.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            log_file_path (str): The path to the log file.
+        """
+        try:
+            # Create a logger with the name of the current module
+            self.logger = logging.getLogger(__name__)
+            # Set the logger level to DEBUG
+            self.logger.setLevel(logging.DEBUG)
+
+            # Define a formatter with the desired log message format
+            formatter = logging.Formatter('%(asctime)s - %(name)s - %(module)s - '
```
Comment (Fr3dMer) [inline src/logging_1.0.py:37]: A few lines here are over 79 char

File: src/logging_1.0.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                                          '%(levelname)s - %(message)s - %(lineno)d',
+                                           datefmt='%y-%m-%d %H:%M:%S')
+
+            # Create a rotating file handler for log files
+            file_handler = RotatingFileHandler(log_file_path, maxBytes=20480, backupCount=5)
+            file_handler.setFormatter(formatter)
+            file_handler.setLevel(logging.DEBUG)
+
+            # Create a stream handler to output log messages to the console
```
Comment (Fr3dMer) [inline src/logging_1.0.py:46]: Comment over 72 char long

File: src/logging_1.0.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            # Log initialization errors
+            print(f"Error during logger initialization: {e}")
+            sys.exit(1)
+
+    def handle_exception(self, exc_type, exc_value, exc_traceback):
+        # Log uncaught exceptions with the logger
+        self.logger.error("Uncaught exception",
+                          exc_info=(exc_type, exc_value, exc_traceback))
+        """
+        Custom exception handler to log uncaught exceptions.
```
Comment (Fr3dMer) [inline src/logging_1.0.py:68]: Should docstring be at start of func?

**Review by Fr3dMer — CHANGES_REQUESTED** [review]
Looks good, just a few pep-8 violations to fix

File: src/logging_1.0.py (hunk 4 above)
Comment (AbiHaddon) [reply src/logging_1.0.py:23]: Amended to PanelAppDB_log_file.log

File: src/logging_1.0.py (hunk 5 above)
Comment (AbiHaddon) [reply src/logging_1.0.py:37]: formatter = logging.Formatter('%(asctime)s - %(name)s - %(module)s - ' = i'm being told this line is 70 characters. Slightly confused..

File: src/logging_1.0.py (hunk 6 above)
Comment (AbiHaddon) [reply src/logging_1.0.py:46]: Amended to 59 characters

File: src/logging_1.0.py (hunk 5 above)
Comment (AbiHaddon) [reply src/logging_1.0.py:37]: The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation. -https://peps.python.org/pep-0008/#maximum-line-length 

Am I not doing this?

**Review by Fr3dMer — APPROVED** [review]
Looks good to me :)

**Pull request merged**